src/EditorFeatures/Core/Implementation/Suggestions/SuggestedAction.cs

```diff
@@ -168,14 +210,19 @@ void IDisposable.Dispose()
             // do nothing
         }
 
-        public virtual IEnumerable<SuggestedActionSet> ActionSets
+        public virtual bool HasActionSets
         {
             get
             {
-                return null;
+                return false;
             }
         }
 
+        public virtual Task<IEnumerable<SuggestedActionSet>> GetActionSetsAsync(CancellationToken cancellationToken)
+        {
+            return SpecializedTasks.Default<IEnumerable<SuggestedActionSet>>();
+        }
```

Comment:
This is an API in the VS platform layer that we implement. I will pass this feedback along to the appropriate folks so that they can add appropriate documentation / guidance for implementers.

The API treats null and empty collection as the same (i.e. there are no action sets).